### DISCUSSION TITLE
Creating a low-code block theme development workflow with WordPress Playground and the Create Block Theme plugin

### DISCUSSION THREAD

**Discussion by ndiego** [WordPress/developer-blog-content#266] [Topic Ideas]
This article will detail how you can set up a low-code (or virtually no-low) development workflow for block themes using [WordPress Playground](https://wordpress.github.io/wordpress-playground/) and the [Create Block Theme](https://wordpress.org/plugins/create-block-theme/) plugin. So long as the theme has a repository on GitHub, you can let contributors submit PRs to modify the theme without touching a line of code. Here’s how it works: 

- The contributor clicks on a link, presumably in the readme of the repo
- A Playground instance is opened up containing the main branch of the theme
- The contributor can then make changes to the theme in the Site Editor
- Once changes are complete, they write those changes to the theme using the Create Block Theme plugin
- Then, using a preset configuration set by the Playground blueprint and the [Query API](https://wordpress.github.io/wordpress-playground/query-api), the contributor can create a PR from within Playground for their changes
- Then the PR is open in the GitHub repo where they can make any additional changes (modify the title, add labels, etc.)

The goal of this article is to show what is currently possible with Playground and inspire readers to both use Playground and explore how it can be used in novel workflows. 

This article's publication will need to wait for the [GitHub proxy](https://github-proxy.com/) to be moved to an official location, which is in the works.

cc @adamziel

**Comment by abhansnuk** [comment]
This will be a really opportune post and of significant help to theme devs as well as showcasing what you can do with WP today. 

if we can give a little extra help (could be by links) to help those follow or who are not as used to GitHub, this would be an additional benefit. 

Can we link to materials on Playground too at Learn.WordPress.org

Thanks Nick.

**Comment by colorful-tones** [comment]
What is this mysterious GitHub Proxy? 🤔 i visited the link and saw what was there, but where does the code live and who sponsors this project? 🕵️

**Reply by ndiego** [reply]
This was spun up as an experiment to see if it was possible. It allows you to circumvent CORS issues, but it would need to be moved to DotOrg somewhere if we are to reference it in this article.

**Comment by bph** [comment]
This topic has been approved. Please continue the conversation on the issue

- #272

**Comment by bph** [comment]
Published: [Streamlining block theme development with WordPress Playground and GitHub](https://developer.wordpress.org/news/2026/01/streamlining-block-theme-development-with-wordpress-playground-and-github/)
